### DISCUSSION TITLE
Home assistant VM on proxmox not starting

### DISCUSSION THREAD

**Discussion by Rvh91** [tteck/Proxmox#3356] [Help Needed]
I followed the guide here:
https://www.derekseaman.com/2023/10/home-assistant-proxmox-ve-8-0-quick-start-guide-2.html

which utilizes the script of this repository
`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/vm/haos-vm.sh)"`
 to create a home assistant VM. I've created both the q35 type as well as the i440fx.

However, both VM's are stuck while booting. I get the:
_[WARN] Home Assistant CLI is not running! Jump into emergency console…_

I've tried the typing 'login' solution that seems to work for some people,
as well as changing the dns settings for the ipv4 interface.

However both dont seem to work. It seems DNS related, as 
`systemctl status docker`

returns errors about DNS resolving.

However, if I create a VM following this guide:
https://forum.proxmox.com/threads/guide-install-home-assistant-os-in-a-vm.143251/

it works out of the box.

Could someone shed some light on this issue?
What is the difference between the VM this script creates vs the guide on the proxmox forums?

**Comment by Mellowlynx** [comment]
Installed a HA VM just yesterday together with a friend for his new house, and I have never seen that error.

Did the script finish correctly?
Also try the Default settings install, If I recall correctly the CACHE is now “Write Trough” by default and that's the only big thing they change during the install in that guide.

**Reply by Mellowlynx** [reply]
Yep, as part of the `function default_settings()` on line 160 you see: `DISK_CACHE="cache=writethrough,"` 
So run with default settings, don't leave the page or close the Shell popup and try again.

If you get an error during install or after, post is here.

**Reply by Rvh91** [reply]
will try with the defaults when i have time!

**Comment by jomud9** [comment]
By chance someone in our local HA group asking for help yesterday and I just free after feeding my cats. :-)

It is quite funny that the script looks awesome on every nodes in my PVE setup, but not on his.

After the troubleshooting session, I have a feeling some parameter can't be push thru.
Under "Emergency console"
- ping, nslookup and other network parameter show it works no problem
- "systemctl status docker" is the key - HA can't be start due to docker missing external "DNS" setting.

As OP said I have the same finding on that VM, there must be glitches.

**Reply by Rvh91** [reply]
same on my node indeed. I was also able to ping google.com etc just fine. Weird thing was that i had tried to create a node with the script earlier. Had the same issue as before, but used the nmcli to set the ipv. 4 dns, and the ipv4.ignore-auto-dns to yes. This did not work after a reboot, left it sit for some time to work on something else. But for some miraculous reason it did work after that. That node however kept having issues with restoring backups etc., so i gave up on it.
Also on the latest node i created with the script it never got past the emergency console.

i will see if using the defaults will work for me. I tried the advanced setting as I was planning to use a google coral for a frigate addon in HAOS, so as i understand i would need a q35 machine type.

**Comment by jomud9** [comment]
Forgot to add. 
HA OS VM creation is running fine on my testing nodes even using default. 

And under the node in trouble, I did not see differences among all different VM settings

I fixed the issue via the suggestion from https://community.home-assistant.io/t/cli-not-starting-on-new-install/317855/41

BUT the tricky part - I just inspect newly working HAOS VM and I found under nmcli con edit "<Your Connection Name>" (for my case, "Supervisor enp0s18")...  ipv4.dns is blink. 

I concluded it is more a HA problem under Proxmox, not tteck script. However it is a good note for us and put that as a workaround.

**Comment by Mellowlynx** [comment]
The VM should get the DNS from the DHCP server, so maybe some funny business on the network part.
I do agree with you that this is a HA problem and not a script problem.

**Comment by Rvh91** [comment]
I noticed just now that in the other guide that does use the script. Some other user (Djones) made the remark that apparently we cant use a q35 system on proxmox for HAS?

"you can’t use q35 on a proxmox system running in legacy bios mode and use OVMF (UEFI) you have to use i440fx. q35 will work on seabios vm just not with HAS"

`https://www.derekseaman.com/2023/10/home-assistant-proxmox-ve-8-0-quick-start-guide-2.html`
